File: Stack & Queue/Java/ValidParenthesis.java
import java.util.*;
public class ValidParenthesis {
    static boolean isValid(String s) {
        Stack<Character> stack = new Stack<Character>();
        for (char c : s.toCharArray()) {
            if (c == '(')
                stack.push(')');
            else if (c == '{')
                stack.push('}');
            else if (c == '[')
                stack.push(']');
            else if (stack.isEmpty() || stack.pop() != c)
                return false;
        }
        return stack.isEmpty();
    }

    public static void main(String[] args) {
        String expr = "()]{}";
        if (isValid(expr))
            System.out.println("Balanced ");
        else
            System.out.println("Not Balanced ");
    }
}

File: Stack & Queue/python/ValidParenthesis.py
def valid_parentheses(s):
    def match(o, c):
        if o == '(' and c == ')':    return True
        if o == '{' and c == '}':    return True
        if o == '[' and c == ']':    return True
        return False
    
    stack = []
    for c in s:
        if c in '({[':    stack.append(c)
        else:
            if not stack:    return False
            if not match(stack.pop(), c):    return False
    return len(stack) == 0

if __name__=='__main__':
    s = '()'
    print(valid_parentheses(s))